app/controllers/api/alert_definitions_controller.rb

```diff
@@ -1,26 +1,31 @@
 module Api
   class AlertDefinitionsController < BaseController
-    REQUIRED_FIELDS = %w(description db expression options).freeze
 
     before_action :set_additional_attributes
 
     def create_resource(type, id, data = {})
       assert_id_not_specified(data, type)
-      assert_all_required_fields_exists(data, type, REQUIRED_FIELDS)
       begin
-        data["expression"] = MiqExpression.new(data["expression"])
+        create_miq_expression(data)
         data["enabled"] = true if data["enabled"].nil?
```

Comment:
Again, I know you're not touching this here (and feel free to leave for a follow up if appropriate) but this is more business logic - "an alert definition is enabled by default" - that belongs in the model

Reply:
Right, removing that too. This should be handled explicitly by the user.
cc @moolitayer @elad661